install_config/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,421 @@
+proxy machine:
+
+====
+----
+# scp /etc/origin/proxy/master/ca.crt \
+      root@proxy.example.com:/etc/pki/CA/certs/
+
+# scp /etc/origin/proxy/proxy.example.com.crt \
+      /etc/origin/proxy/authproxy.pem \
+      root@proxy.example.com:/etc/pki/tls/certs/
+
+# scp /etc/origin/proxy/proxy.example.com.key \
+      root@proxy.example.com:/etc/pki/tls/private/
+----
+====
+
+[[phase-2-step-2-sssd-configuration]]
+=== Step 2: SSSD Configuration
+Install a new VM with a recent operating system. Run SSSD 1.12.0 or later so
+that you can use the *mod_identity_lookup* module. The examples in this topic
+use a Red Hat Enterprise Linux 7.2 Server.
+
+Install all of the necessary dependencies:
+
+====
+----
+# yum install -y sssd \
+                 sssd-dbus \
+                 realmd \
+                 httpd \
+                 mod_session \
+                 mod_ssl \
+                 mod_lookup_identity \
+                 mod_authnz_pam
+----
+====
+
+This gives you the needed SSSD and the web server components. Now, set up SSSD
+to authenticate this VM against the LDAP server. If the LDAP server is a FreeIPA
+or Active Directory environment, then *realmd* can be used to join this machine
+to the domain.
+
+====
+----
+# realm join ldap.example.com
+----
+====
+
+If you are not running a domain, then use the *authconfig* tool:
+
+====
+----
+# authconfig --update --enablesssd --enablesssdauth \
+             --ldapserver=ldap.example.com \
+             --enableldaptls \
+             --ldaploadcert=http://ldap.example.com/ca.crt
+----
+====
+
+This will create *_/etc/sssd/sssd.conf_* with most of the appropriate settings.
+
+[NOTE]
+====
+In RHEL 7, *authconfig* does not create the *_/etc/openldap/cacerts_* directory,
+so you may need to create it manually before running the above command.
+====
+
+If you want to use SSSD to manage failover situations for LDAP, this can be
+configured by adding additional entries in *_/etc/sssd/sssd.conf_* on the
+*ldap_uri* line. Systems enrolled with FreeIPA will automatically handle
+failover using DNS SRV records.
+
+Restart SSSD to ensure that all of the changes are applied properly:
+
+====
+----
+$ systemctl restart sssd.service
+----
+====
+
+Test that the user information can be retrieved properly:
+
+====
+----
+$ getent passwd <username>
+username:*:12345:12345:Example User:/home/username:/usr/bin/bash
+----
+====
+
+Attempt to log into the VM as an LDAP user and confirm that the authentication
+is properly set up. This can be done via the local console or a remote service
+such as SSH.
+
+[[phase-2-step-3-apache-configuration]]
+=== Step 3: Apache Configuration
+
+You need to set up Apache to communicate with SSSD. First, create a PAM stack
+file for use with Apache. Create the *_/etc/pam.d/openshift_* file and add the
+following contents:
+
+====
+----
+auth required pam_sss.so
+account required pam_sss.so
+----
+====
+
+This will tell PAM (the pluggable authentication module) that when an
```

Comment:
"tell" is very casual language, perhaps too casual. Avoid future tense. Suggested edit:

This configuration enables PAM (the pluggable authentication module) to use *pam_sss.so* to determine authentication and access control when an authentication request is issued for the *openshift* stack.